### DISCUSSION TITLE
Node sync between maps

### DISCUSSION THREAD

**Discussion by olazzzzz** [freeplane/freeplane#1024] [Ideas]
Hi everyone,

Is it already possible (or is there a way to implement) node sync in different mindmaps?
So that if you would update a node on Mindmap X, it would automatically update on Mindmap Z?
From what I managed to search, I did not find anything on this. 
It could be quite convenient in my opinion, in situations where several projects/mindmaps intersect in some topics.
Thank you!

**Comment by mrexpertise** [comment]
it's a pity that there are no replies to this. I am also looking for a way to implement it in my mindmaps so I don't have to update the same data in multiple maps. anyone has any ideas how to do this?

**Reply by macmarrum** [reply]
You can use a NodeChangeListener, which would react to changes in a node. 
The node could have an attribute pointing to the mind map and node that it should sync with.
The code of NodeChangeListener would then open the mind map and update the target node.

Please see https://github.com/freeplane/freeplane/discussions/1275#discussioncomment-6327975

**Reply by macmarrum** [reply]
You can also have a look at a similar topic but within a mind map:
- https://github.com/freeplane/freeplane/discussions/1297

**Comment by olazzzzz** [comment]
Thank you both!
@macmarrum could you expand on that please? I have installed the script and add on, but this part "The node could have an attribute pointing to the mind map and node that it should sync with." is the one I don't know how to do.
Thanks for your help.

**Reply by macmarrum** [reply]
As far as I know, there is no readily available solution for your problem. But I can imagine a way to implement it using the Scripting API. To do that, you might take some ideas from the solutions for similar problems and the approach I proposed, but in the end you'll need to write the relevant script(s) yourself.

You can study [syncFollowerBranchesWithMasterBranches.groovy](https://raw.githubusercontent.com/macmarrum/freeplane-scripting/main/src/scripts/syncFollowerBranchesWithMasterBranches.groovy). It's a solution to the problem of syncing a branch within a mind map. It uses the approach of storing target node's reference in an attribute. This is what I had in mind when I wrote "The node could have an attribute pointing to the mind map and node that it should sync with."

Node sync between mind maps is quite a task, requiring good knowledge of Freeplane, its Scripting API and being comfortable coding in Groovy or Java, but I think it's achievable. Please feel free to report back if you get stuck – I'll try to help. That is, if you decide to take in on.
